src/Helpers/JWKFetcher.php

```diff
@@ -38,43 +45,69 @@ public function __construct(CacheHandler $cache = null, $guzzleOptions = [])
     }
 
     /**
+     * Convert a certificate to PEM.
+     *
+     * @param string $cert Certificate to convert.
      *
-     * @param  string $cert
      * @return string
      */
     protected function convertCertToPem($cert)
     {
-        return '-----BEGIN CERTIFICATE-----'.PHP_EOL.chunk_split($cert, 64, PHP_EOL).'-----END CERTIFICATE-----'.PHP_EOL;
+        $output  = '-----BEGIN CERTIFICATE-----'.PHP_EOL;
+        $output .= chunk_split($cert, 64, PHP_EOL);
+        $output .= '-----END CERTIFICATE-----'.PHP_EOL;
+        return $output;
     }
 
     /**
+     * Fetch JWKS from the token issuer.
      *
-     * @param string $iss
+     * @param string      $domain    Typically the issuer for the JWKS being fetched; include a trailing slash.
+     * @param null|string $jwks_path Path to the JWKS from the issuer domain; pass null to use $domain as the URL.
      *
      * @return array|mixed|null
-     *
-     * @throws \Exception
      */
-    public function fetchKeys($iss)
+    public function fetchKeys($domain, $jwks_path = '.well-known/jwks.json')
     {
-        $url = "{$iss}.well-known/jwks.json";
+        // If the path parameter is empty, use the $domain parameter as a complete URL.
+        if (empty( $jwks_path )) {
+            $domain_parsed = parse_url( $domain );
+            $domain        = $domain_parsed['scheme'].'://'.$domain_parsed['host'];
+            $domain       .= empty( $domain_parsed['port'] ) ? '' : ':'.$domain_parsed['port'];
+            $domain       .= '/';
+            $jwks_path     = empty( $domain_parsed['path'] ) ? '' : substr( $domain_parsed['path'], 1 );
```

Comment:
what's the difference between appending a `""` and an empty `$domain_parsed['path']` ?

Anyway.. I think this logic is too complex for the sake of just using a URI. I'd do this steps:

1. check that a scheme is present in `$domain`. If not, use `https`
2. check that `$jwks_path` is empty/null. If not, append it to the `$domain`
3. make the request to `$domain` (which could have been modified on step 2)

Reply:
> check that a scheme is present in $domain. If not, use https

No need. That always expected a valid URL and should continue to do so. 

> check that $jwks_path is empty/null. If not, append it to the $domain

That's what I was doing before. This complete URL support was based on your suggestion, which I liked :)

> make the request to $domain (which could have been modified on step 2)

API client relies on a `domain` and `basePath`, hence the logic here.